### ISSUE TITLE
Assignment 9.1

### ISSUE THREAD

**Issue by awilli1186** [theironyard-frontend-nashville/assignments#66]
https://github.com/awilli1186/assignment-9.1

**Comment by egdelwonk** [comment]
You have a few syntax errors. Can you correct them?
- You have an extra closing `}` on line 89. 

Also, you are checking `if(options)` multiple times, you can reduce that check so that you only check options once, and add the remaining conditionals inside of that if block, or do the checks like `if(options && options.hasOwnProperty('color'))`

Please adjust these issues and resubmit.

**Comment by awilli1186** [comment]
the closing tag on line 89 is the closing tag for the Human constructor.
I'm not sure I understand what you are wanting me to do with the
if(options) stuff.

On Tue, Jul 21, 2015 at 9:34 PM, William Golden notifications@github.com
wrote:

> You have a few syntax errors. Can you correct them?
> - You have an extra closing } on line 89.
> 
> Also, you are checking if(options) multiple times, you can reduce that
> check so that you only check options once, and add the remaining
> conditionals inside of that if block, or do the checks like if(options &&
> options.hasOwnProperty('color'))
> 
> Please adjust these issues and resubmit.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/theironyard-frontend-nashville/assignments/issues/66#issuecomment-123538899
> .

## 

_Adam Williams_
awilli1186@gmail.com

**Comment by egdelwonk** [comment]
Oh, maybe you don't have any syntax errors, but the indention levels make it really hard to follow. 

Can you clean / realign the code indention levels to make it easier to read?

Keeping a consistent indention level is really important for other programmers to read your code. :horse:

**Comment by egdelwonk** [comment]
As for the options stuff:

You are checking `if(options)` three times:

``` javascript
if (options) {
    if (options.hasOwnProperty('hungry')){
      console.log("I'm true");
      this.hungry = options.hungry;
    }

    if (options) {
      if (options.hasOwnProperty('color')){
        this.color = options.color;
      }

      if (options) {
        if (options.hasOwnProperty('status')){
          this.status = options.status;
        }
      }
  };
}
```

You can change this to be a bit easier to read, and less code:

``` javascript
if (options) {
  if (options.hasOwnProperty('hungry')){
    console.log("I'm true");
    this.hungry = options.hungry;
  }


  if (options.hasOwnProperty('color')){
    this.color = options.color;
  }

  if (options.hasOwnProperty('status')){
    this.status = options.status;
  }
}
```

or

``` javascript
if (options && options.hasOwnProperty('hungry')){
  console.log("I'm true");
  this.hungry = options.hungry;
}


if (options.hasOwnProperty('color')){
  this.color = options.color;
}

if (options && options.hasOwnProperty('status')){
  this.status = options.status;
}
```

**Comment by awilli1186** [comment]
ok.  I think I understand. I didn't know you could nest the if statements
like that. I will get it fixed and the indention corrected.

Man this week I am feeling not very smart. I am very frustrated right now
with this stuff.

On Tue, Jul 21, 2015 at 10:07 PM, William Golden notifications@github.com
wrote:

> As for the options stuff:
> 
> You are checking if(options) three times:
> 
> if (options) {
>     if (options.hasOwnProperty('hungry')){
>       console.log("I'm true");
>       this.hungry = options.hungry;
>     }
> 
> ```
> if (options) {
>   if (options.hasOwnProperty('color')){
>     this.color = options.color;
>   }
> 
>   if (options) {
>     if (options.hasOwnProperty('status')){
>       this.status = options.status;
>     }
>   }
> ```
> 
>   };
> }
> 
> You can change this be a bit easier to read, and less code:
> 
> if (options) {
>   if (options.hasOwnProperty('hungry')){
>     console.log("I'm true");
>     this.hungry = options.hungry;
>   }
> 
>   if (options.hasOwnProperty('color')){
>     this.color = options.color;
>   }
> 
>   if (options.hasOwnProperty('status')){
>     this.status = options.status;
>   }
> }
> 
> or
> 
> if (options && options.hasOwnProperty('hungry')){
>   console.log("I'm true");
>   this.hungry = options.hungry;
> }
> 
> if (options.hasOwnProperty('color')){
>   this.color = options.color;
> }
> if (options && options.hasOwnProperty('status')){
>   this.status = options.status;
> }
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/theironyard-frontend-nashville/assignments/issues/66#issuecomment-123548987
> .

## 

_Adam Williams_
awilli1186@gmail.com

**Comment by egdelwonk** [comment]
I think you're doing a good job. Try to keep perspective. You are learning a ton of stuff all at once. Keep fighting the good fight! :fire_engine: :fire:

**Comment by awilli1186** [comment]
thanks. I am trying super hard to keep this in perspective. Its difficult
when I want to be good at this right now, but it isn't completely clicking.
I am sure I haven't had to learn this much this quickly in my entire life.
I think my brain is just overloaded with information.

On Tue, Jul 21, 2015 at 10:15 PM, William Golden notifications@github.com
wrote:

> I think you're doing a good job. Try to keep perspective. You are learning
> a ton of stuff all at once. Keep fighting the good fight! [image:
> :fire_engine:] [image: :fire:]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/theironyard-frontend-nashville/assignments/issues/66#issuecomment-123550514
> .

## 

_Adam Williams_
awilli1186@gmail.com

**Comment by egdelwonk** [comment]
The course is designed to keep you right outside of your comfort zone. This is where learning happens:

![image](https://cloud.githubusercontent.com/assets/512548/8817321/b482b8e8-2ff7-11e5-88b7-b776e3c746f4.png)

**Comment by awilli1186** [comment]
Will, I went through and made the updates.
